awx/main/tasks/facts.py

```diff
@@ -34,31 +35,36 @@ def start_fact_cache(hosts, destination, log_data, timeout=None, inventory_id=No
     if timeout is None:
         timeout = settings.ANSIBLE_FACT_CACHE_TIMEOUT
 
-    last_filepath_written = None
+    last_write_time = None
+
     for host in hosts:
-        hosts_cached.append(host)
+        hosts_cached.append(host.name)
         if not host.ansible_facts_modified or (timeout and host.ansible_facts_modified < now() - datetime.timedelta(seconds=timeout)):
             continue  # facts are expired - do not write them
 
-        filepath = os.sep.join(map(str, [destination, host.name]))
+        filepath = os.path.join(destination, host.name)
         if not os.path.realpath(filepath).startswith(destination):
-            system_tracking_logger.error('facts for host {} could not be cached'.format(smart_str(host.name)))
+            system_tracking_logger.error(f'facts for host {smart_str(host.name)} could not be cached')
             continue
 
         try:
             with codecs.open(filepath, 'w', encoding='utf-8') as f:
                 os.chmod(f.name, 0o600)
                 json.dump(host.ansible_facts, f)
                 log_data['written_ct'] += 1
-                last_filepath_written = filepath
+                last_write_time = os.path.getmtime(filepath)
         except IOError:
-            system_tracking_logger.error('facts for host {} could not be cached'.format(smart_str(host.name)))
+            system_tracking_logger.error(f'facts for host {smart_str(host.name)} could not be cached')
             continue
 
-    if last_filepath_written:
-        return os.path.getmtime(last_filepath_written), hosts_cached
-
-    return None, hosts_cached
+    # Write summary file to artifacts dir
+    if inventory_id is not None:
+        artifact_dir = os.path.join('artifacts', f'inventory_{inventory_id}')
```

Comment:
Artifacts needs to be relative to the job private_data_dir. This is nonsense path.